bson/src/main/org/bson/internal/ChildCodecRegistry.java

```diff
@@ -20,36 +20,64 @@
 import org.bson.codecs.Codec;
 import org.bson.codecs.configuration.CodecRegistry;
 
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static org.bson.assertions.Assertions.isTrueArgument;
+import static org.bson.assertions.Assertions.notNull;
+
 // An implementation of CodecRegistry that is used to detect cyclic dependencies between Codecs
 class ChildCodecRegistry<T> implements CodecRegistry {
 
     private final ChildCodecRegistry<?> parent;
     private final CycleDetectingCodecRegistry registry;
     private final Class<T> codecClass;
+    private final List<Type> types;
```

Comment:
Suggestion: Annotate as nullable - so its clear in the code.

Also in LazyCodec and the cache key

Reply:
Apologies - see this was merged

Reply:
No worries.  I mentioned in another thread that we don't have Nullable annotations available in the bson module.  I think we already have a ticket for that.